client/my-sites/email/email-management/home/email-mailbox-warnings.jsx

```diff
@@ -0,0 +1,167 @@
+} from 'calypso/lib/emails/email-provider-constants';
+import { isEmailForwardAccount } from 'calypso/lib/emails/is-email-forward-account';
+
+const MailboxListItemWarning = ( { warningText } ) => {
+	return (
+		<div className="email-mailbox-warnings__warning">
+			<Gridicon icon="info-outline" size={ 18 } />
+			<span>{ warningText }</span>
+		</div>
+	);
+};
+
+const MailboxListItemAction = ( { buttonText, external, ...otherProps } ) => {
+	return (
+		<div className="email-mailbox-warnings__action">
+			<Button compact { ...otherProps }>
+				<span>{ buttonText }</span>
+				{ external && <Gridicon icon="external" /> }
+			</Button>
+		</div>
+	);
+};
+
+const resendEmailForwardVerification = ( mailbox, dispatch ) => {
+	const destination = getEmailForwardAddress( mailbox );
+	dispatch(
+		recordTracksEvent(
+			'calypso_email_management_email_forwarding_resend_verification_email_click',
+			{
+				destination,
+				domain_name: mailbox.domain,
+				mailbox: mailbox.mailbox,
+			}
+		)
+	);
+	dispatch( resendVerificationEmail( mailbox.domain, mailbox.mailbox, destination ) );
+};
+
+const getDetailsForWarning = ( { account, dispatch, mailbox, translate, warning } ) => {
+	const warningSlug = warning.warning_slug;
+
+	if ( isGoogleEmailAccount( account ) ) {
+		if ( warningSlug === EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS ) {
+			const finishSetupForGoogle = {
+				actionProps: {
+					buttonText: translate( 'Finish setup' ),
+					external: true,
+					href: getGoogleAdminUrl( mailbox.domain ),
+					target: '_blank',
+				},
+				warningText: translate( 'Action required' ),
+			};
+
+			if ( isEmailUserAdmin( mailbox ) ) {
+				return finishSetupForGoogle;
+			}
+
+			// For non-admin users we disable the button if the account has been suspended
+			// due to not accepting the account ToS.
+			const googleAccountSuspendedDueToTOS = hasGoogleAccountTOSWarning( account );
+
+			return {
+				actionProps: {
+					...finishSetupForGoogle.actionProps,
+					disabled: googleAccountSuspendedDueToTOS,
+					href: googleAccountSuspendedDueToTOS ? null : finishSetupForGoogle.actionProps.href,
+				},
+				warningText: finishSetupForGoogle.warningText,
+			};
+		}
+
+		return null;
+	}
+
+	if ( isEmailForwardAccount( account ) ) {
+		if ( warningSlug === EMAIL_WARNING_SLUG_UNVERIFIED_FORWARD ) {
+			return {
+				actionProps: {
+					buttonText: translate( 'Resend verification email' ),
+					onClick: () => resendEmailForwardVerification( mailbox, dispatch ),
+				},
+				warningText: translate( 'Verification required' ),
+			};
+		}
+
+		return null;
+	}
+
+	return null;
+};
+
+const EmailMailboxWarning = ( { warningActionProps, warningText } ) => {
+	if ( ! warningActionProps ) {
+		return <MailboxListItemWarning warningText={ warningText } />;
+	}
+
+	return (
+		<>
+			<MailboxListItemWarning warningText={ warningText } />
+			{ warningActionProps && <MailboxListItemAction { ...warningActionProps } /> }
+		</>
+	);
```

Comment:
Apologies if I've missed something but I feel like the first condition is not needed:

```suggestion
	return (
		<>
			<MailboxListItemWarning warningText={ warningText } />
			{ warningActionProps && <MailboxListItemAction { ...warningActionProps } /> }
		</>
	);
```

Reply:
🤦 I fixed the second block of logic, but didn't remove the initial check. Thanks for catching this!